src/api/http.go

```diff
@@ -350,121 +375,121 @@ func newServerMux(c muxConfig, gateway Gatewayer, csrfStore *CSRFStore, rpc *web
 	//  dst: Destination address
 	//  Returns total amount spent if successful, otherwise error describing
 	//  failure status.
-	webHandlerV1("/wallet/spend", walletSpendHandler(gateway))
+	webHandlerV1("/wallet/spend", forAPISet(walletSpendHandler(gateway), APIWallet))
 
 	// Creates a transaction from a wallet
-	webHandlerV1("/wallet/transaction", createTransactionHandler(gateway))
+	webHandlerV1("/wallet/transaction", forAPISet(createTransactionHandler(gateway), APIWallet))
 
 	// GET Arguments:
 	//      id: Wallet ID
 	// Returns all pending transanction for all addresses by selected Wallet
-	webHandlerV1("/wallet/transactions", walletTransactionsHandler(gateway))
+	webHandlerV1("/wallet/transactions", forAPISet(walletTransactionsHandler(gateway), APIWallet))
 
 	// Update wallet label
 	// POST Arguments:
 	//     id: wallet id
 	//     label: wallet label
-	webHandlerV1("/wallet/update", walletUpdateHandler(gateway))
+	webHandlerV1("/wallet/update", forAPISet(walletUpdateHandler(gateway), APIWallet))
 
 	// Returns all loaded wallets
 	// returns sensitive information
-	webHandlerV1("/wallets", walletsHandler(gateway))
+	webHandlerV1("/wallets", forAPISet(walletsHandler(gateway), APIWallet))
 
 	// Returns wallets directory path
-	webHandlerV1("/wallets/folderName", getWalletFolder(gateway))
+	webHandlerV1("/wallets/folderName", forAPISet(getWalletFolder(gateway), APIWallet))
 
 	// Generate wallet seed
 	// GET Arguments:
 	//     entropy: entropy bitsize.
-	webHandlerV1("/wallet/newSeed", newWalletSeed(gateway))
+	webHandlerV1("/wallet/newSeed", forAPISet(newWalletSeed(gateway), APIWallet))
 
 	// Gets seed of wallet of given id
 	// GET Arguments:
 	//     id: wallet id
 	//     password: wallet password
-	webHandlerV1("/wallet/seed", walletSeedHandler(gateway))
+	webHandlerV1("/wallet/seed", forAPISet(walletSeedHandler(gateway), APISeed)) // FIXME: APIWallet?
```

Comment:
What was the old behaviour? Did `-enable-seed-api`  require `-enabled-wallet-api`?

Reply:
Both options are independent , but this handler checks for both flags set .